### PR TITLE
feat: added party type to defendant details case field (CMC-625)

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-625

### Change description ###

- added party type to defendant details case field
- new element was added to Party complex type - this was required because of the way field interpolation works

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-definition/ComplexTypes/Party.json [hunk 1]
```diff
@@ -109,5 +109,14 @@
     "ElementLabel": "Party name",
     "FieldShowCondition": "type = \"DO_NOT_SHOW\"",
     "SecurityClassification": "Public"
+  },
+  {
+    "ID": "Party",
+    "ListElementCode": "typeDisplayValue",
```
Comment (AlisonRevittHM) [inline ccd-definition/ComplexTypes/Party.json]: @pd-dudkiewicz - as discussed, please can we change this to partyTypeDisplayValue?

File: ccd-definition/ComplexTypes/Party.json [hunk 2]
```diff
@@ -109,5 +109,14 @@
...
+    "FieldType": "Text",
+    "ElementLabel": "Party type displayed value",
+    "FieldShowCondition": "type = \"DO_NOT_SHOW\"",
+    "SecurityClassification": "Public",
+    "_Comments": "Field interpolation used on fixed list field will display its code instead of text, hence the workaround field populated on claim creation."
```
Comment (AlisonRevittHM) [inline ccd-definition/ComplexTypes/Party.json]: @pd-dudkiewicz can we also add _Category WORKAROUND?

**Review by AlisonRevittHM — CHANGES_REQUESTED** [review]
data review

File: ccd-definition/ComplexTypes/Party.json (hunk 1 above)
Comment (pd-dudkiewicz) [reply ccd-definition/ComplexTypes/Party.json]: done

File: ccd-definition/ComplexTypes/Party.json (hunk 2 above)
Comment (pd-dudkiewicz) [reply ccd-definition/ComplexTypes/Party.json]: done

**Review by douglasrice — APPROVED** [review]

**Review by sabahirfan — APPROVED** [review]

**Pull request merged**